pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,136 @@
+package agent
+
+import (
+	//corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	BareMetal *BareMetal `json:"bareMetal"` // mandatory for now, possibly optional in future
```

Comment:
let's create a Provider field and then add BareMetal to it. this should allow for extension later if we need different provider for agent based installs.